### PR TITLE
IR-3984 viewer transition problems

### PR DESCRIPTION
**Pull request by dinomut1**
Fixes for bugs observed when switching between engine viewer canvases in the wizard. 

Pointer entities are generated in ClientInputHooks which get lost when viewer canvas gets unmounted. Since this data is only tracked in a one way reference from the PointerComponent to the camera entity associated with a viewer, the simplest fix is to detect stale pointers by checking if their camera entity still exists each frame

Also, the initialization of the viewer and of the origin have been split apart so that the scene doesn't need to be unloaded and reloaded every time a canvas is created or destroyed. Instead the base spatial environment is initialized on its own in initializeSpatialEngine, and a new function, initializeSpatialViewer, initializes the viewer. This way we can create and destroy viewers and preserve the underlying scene

Closes https://tsu.atlassian.net/browse/IR-3984

### CODE REVIEW

**Review by SamMazerIR — APPROVED** [review]

File: packages/editor/src/components/EditorContainer.tsx [hunk 1]
```diff
@@ -168,12 +169,19 @@ const EditorContainer = () => {
     }
   }, [scenePath.value])
 
-  const viewerEntity = useMutableState(EngineState).viewerEntity.value
+  useEffect(() => {
+    initializeSpatialEngine()
+    return () => {
+      destroySpatialEngine()
+    }
+  }, [])
```
Comment (Rezmason) [inline packages/editor/src/components/EditorContainer.tsx:177]: I'd like to better understand how often this `useEffect` is run; it has no dependencies, but it initializes and destroys some things that look like a big deal, so to my undertrained eye it looks conspicuous

**Review by Rezmason — APPROVED** [review]

File: packages/editor/src/components/EditorContainer.tsx (hunk 1 above)
Comment (Rezmason) [reply packages/editor/src/components/EditorContainer.tsx:177]: "Runs only once, when the editor container is mounted."

File: packages/spatial/src/input/systems/ClientInputSystem.tsx [hunk 2]
```diff
@@ -106,9 +107,16 @@ const execute = () => {
     getMutableComponent(eid, InputComponent).inputSources.set([])
   }
 
+  const stalePointers: Entity[] = []
+
   // update 2D screen-based (driven by pointer api) input sources
   for (const eid of pointersQuery()) {
     const pointer = getComponent(eid, InputPointerComponent)
+    // check if pointer camera entity still exists
```
Comment (heysokam) [inline packages/spatial/src/input/systems/ClientInputSystem.tsx:115]: This PR will be merged soon: https://github.com/ir-engine/ir-engine/pull/26
These new `stalePointers` blocks of code are not tested, but probably should have specific cases created for them :writing_hand:

**Review by HexaField — APPROVED** [review]

File: packages/spatial/src/input/systems/ClientInputSystem.tsx (hunk 2 above)
Comment (heysokam) [reply packages/spatial/src/input/systems/ClientInputSystem.tsx:115]: I will update #26 then, since this was merged first into dev

**Pull request merged**